### PR TITLE
Automation tool for benchmark framework

### PR DESCRIPTION
**Pull request by realwanpengli**
Simple automation tool for benchmark.
Only support Linux for now.

### CODE REVIEW

File: SignalRServiceBenchmarkPlugin/utils/Commander/ArgsOption.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{
+    // TODO: edit descriptions and default values
+    public class ArgsOption
+    {
+        // App server
+        [Option("AppserverPort", Required = false, Default = 5050, HelpText = "Port of app server.")]
+        public int AppserverPort { get; set; }
+
+        // Remote
+        [Option("RpcPort", Required = false, Default = 5050, HelpText = "Port to be conencted from remote hosts.")]
```
Comment (ccic) [inline SignalRServiceBenchmarkPlugin/utils/Commander/ArgsOption.cs]: Default RpcPort and AppServerPort should be different.

Comment (realwanpengli) [reply SignalRServiceBenchmarkPlugin/utils/Commander/ArgsOption.cs]: Ok, thanks... I forget to change the default value.

File: SignalRServiceBenchmarkPlugin/utils/Commander/Program.cs [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+            var slaveExecutable = PubishExcutable(argsOption.SlaveProject, _baseName);
+
+            //// DEBUG
+            //var baseName = "publish";
+            //var appserverExecutable = new FileInfo(Path.Combine(argsOption.AppserverProject, $"{baseName}.zip"));
+            //var masterExecutable = new FileInfo(Path.Combine(argsOption.MasterProject, $"{baseName}.zip"));
+            //var slaveExecutable = new FileInfo(Path.Combine(argsOption.SlaveProject, $"{baseName}.zip"));
+
+            // Copy executables and configurations
+            CopyExcutables(argsOption.Username, argsOption.Password,
```
Comment (ccic) [inline SignalRServiceBenchmarkPlugin/utils/Commander/Program.cs]: There are too many arguments in this function. Please pass "argsOption" instead.

Comment (realwanpengli) [reply SignalRServiceBenchmarkPlugin/utils/Commander/Program.cs]: I refactor the code, please take a look.

File: SignalRServiceBenchmarkPlugin/utils/Commander/Program.cs [hunk 3]
```diff
@@ -22,242 +22,8 @@ public static void Main(string[] args)
...
-            var publish = Command.Run("dotnet", $"publish -o {baseName}".Split(' '), o => o.WorkingDirectory(projectPath));
-            publish.Wait();
-            if (!publish.Result.Success) throw new Exception(publish.Result.StandardOutput);
-
-            var zip = Command.Run("zip", new string[] { "-r", $"{baseName}.zip", $"{baseName}/" }, o => o.WorkingDirectory(projectPath));
-            zip.Wait();
-            if (!zip.Result.Success) throw new Exception(zip.Result.StandardOutput);
-
-            return new FileInfo(Path.Combine(projectPath, $"{baseName}.zip"));
+            var automationTool = new AutomationTool(argsOption);
```
Comment (realwanpengli) [inline SignalRServiceBenchmarkPlugin/utils/Commander/Program.cs]: Refactor the code, please take a look @ccic

Comment (ccic) [reply SignalRServiceBenchmarkPlugin/utils/Commander/Program.cs]: Yes, it looks clean.

File: SignalRServiceBenchmarkPlugin/utils/Commander/ArgsOption.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        [Option("SlaveList", Required = true, Separator = ',',  HelpText = "Target hosts to connect.")]
+        public IList<string> SlaveList { get; set; }
+
+        [Option("MasterHostname", Required = true, HelpText = "Hostname of master.")]
+        public string MasterHostname { get; set; }
+
+        [Option("AppServerHostname", Required = true, HelpText = "Hostname of app server.")]
+        public string AppServerHostname { get; set; }
+
+        [Option("Username", Required = true, HelpText = "Username of slaves.")]
```
[the remark this answers is not in the document]
Comment (realwanpengli) [reply SignalRServiceBenchmarkPlugin/utils/Commander/ArgsOption.cs]: Ok, thanks.

**Pull request merged**